### PR TITLE
Feat/move app to func

### PR DESCRIPTION
**Pull request by ProfOak**

### CODE REVIEW

File: tests/conftest.py [hunk 1]
```diff
@@ -134,12 +134,10 @@ def teardown():
...
-    _app.config.from_object("sheepdog.test_settings")
+    _app = app_init()
 
     request.addfinalizer(teardown)
 
-    app_init(_app)
-
     _app.logger.setLevel(os.environ.get("GDC_LOG_LEVEL", "WARNING"))
 
     return _app
```
Comment (jbarno) [inline tests/conftest.py]: FYI this fixture is functionally scoped so it is both writing to a tmp file and starting and signpost process with any test that references this.

This problem is excaerbated because the pytest-flask fixture client should depend on app. Not part of this PR, but it is why the API tests were so slow 6 months ago.

File: tests/conftest.py [hunk 2]
```diff
@@ -234,7 +232,7 @@ def build_header(path, method, role='submitter'):
 
 @pytest.fixture(scope='session')
 def dictionary_setup():
```
Comment (jbarno) [inline tests/conftest.py]: I think this fixture should actually take in the app fixture. That way you don't have to modify all the test calls. And it will force you to make the app fixture session scoped.

Comment (ProfOak) [reply tests/conftest.py]: It does not. I tried ):

Comment (jbarno) [reply tests/conftest.py]: Right, was the error was mixing scope of fixtures? How can something that runs once per test session take something that runs once every test?
